src/extension.ts

```diff
@@ -145,6 +158,106 @@ async function copyConnectionString(node: IAzureNode<MongoAccountTreeItem | DocD
+		//tslint:disable:no-non-null-assertion
+		result = await collectionTreeItem!.executeCommand('insertMany', [JSON.stringify(documents)]);
+	} else {
+		result = await insertDocumentsIntoDocdb(<IAzureParentNode<DocDBCollectionTreeItem>>collectionNode, documents, nodes);
+	}
+	await collectionNode.refresh();
+	await vscode.window.showInformationMessage(result);
+}
+
+async function askForDocuments(): Promise<vscode.Uri[]> {
+	let files: vscode.Uri[] = await vscode.workspace.findFiles("*.json");
+	let jsonDocuments: (vscode.QuickPickItem & { uri: vscode.Uri })[] = [];
+	let items: (vscode.QuickPickItem & { uri: vscode.Uri })[] = files.map(file => {
+		return { uri: file, label: vscode.workspace.asRelativePath(file) };
+	});
+	let pickAgain: string = "Pick again";
+	let discontinue = "Discontiue import";
+	while (!jsonDocuments.length) {
+		jsonDocuments = await ext.ui.showQuickPick(items, { canPickMany: true, placeHolder: "Choose a document to upload. Hit Escape to Cancel" });
+		if (!jsonDocuments.length) {
+			let action: string = await vscode.window.showWarningMessage("No document picked. Want to pick again?", pickAgain, discontinue);
+			if (action === discontinue) {
+				throw new UserCancelledError();
+			}
+		}
+	}
+	return jsonDocuments.map(choice => choice.uri);
+}
+
+// tslint:disable-next-line:no-any
+async function parseDocumentsForErrors(nodes: vscode.Uri[]): Promise<any[]> {
+	const parseResult = await parseDocuments(nodes);
+	const documents = parseResult[0];
+	const errors: string[] = parseResult[1];
+	if (errors.length > 0) {
+		ext.outputChannel.show();
+		throw new Error(`Errors found in the following documents: ${errors.join(',')}.\nPlease fix these and try again.`);
+	}
+	return documents;
+}
+
+// tslint:disable-next-line:no-any
+async function parseDocuments(nodes: vscode.Uri[]): Promise<[any[], string[]]> {
+	let documents = [];
+	let errors = {};
+	for (let node of nodes) {
+		const document = (await vscode.workspace.openTextDocument(node));
+		const text = document.getText();
+		let parsed;
+		try {
+			parsed = JSON.parse(text);
+		} catch (e) {
+			const err = parseError(e);
+			const fileName = node.path.split('/').pop();
+			errors[fileName] = err;
+			ext.outputChannel.appendLine(`${fileName}:\n${err}`);
+			await vscode.window.showTextDocument(document);
+		}
+		if (parsed) {
+			if (Array.isArray(parsed)) {
+				documents = documents.concat(parsed);
+			} else {
+				documents.push(parsed);
+			}
+		}
+	}
+	return [documents, Object.keys(errors)];
+}
+
+// tslint:disable-next-line:no-any
+async function insertDocumentsIntoDocdb(collectionNode: IAzureParentNode<DocDBCollectionTreeItem>, documents: any[], nodes: vscode.Uri[]): Promise<string> {
+	let result;
+	let ids = [];
+	const collectionTreeItem = (<DocDBCollectionTreeItem>collectionNode.treeItem);
+	const documentsTreeItem: DocDBDocumentsTreeItem = <DocDBDocumentsTreeItem>(await collectionTreeItem.loadMoreChildren(collectionNode, false))[0];
```

Comment:
loadMoreChildren is not meant to be called directly and could have weird effects on the tree. Call `collectionNode.getCachedChildren` instead. I am refactoring the tree to hopefully make this more clear (but don't worry about the refactor for now)